### ISSUE TITLE
Low AR buff based on Star Rating

### ISSUE THREAD

**Issue by MBmasher** [ppy/osu-performance#65]
(original idea proposed by Nymfaye)

**The Problem**
Currently, most of the playerbase can agree that EZ/low AR is underweighted due to object density/reading not being taken into account into the system.

**The Basic Approach**
Nymfaye proposed that low AR could be buffed based on Star Rating. One could say that this doesn't address the real issue with low AR, but my argument to that is that Star Rating is very closely linked to a map's object density.
I built up my own idea from Nymfaye's idea. I created a low AR threshold, which is based off a map's star rating. Then, I created a bonus which was based on how much lower the AR of the map was compared to the low AR threshold.
[Here's (badly) drawn graphs showing my idea.](https://i.imgur.com/WyXnRf1.png)
For aim pp: the bonus applies whether hidden is enabled or not. However, hidden gives an extra 30% bonus.
For speed pp: the bonus applies only when hidden is enabled.

**Juicy Math**
_Formulas_
[Formula for Low AR Threshold Calculation](https://i.imgur.com/8gymbOo.gif)
- StarRating refers to the separate aim/speed star rating, not the total star rating.

[Formula for Bonus Calculation](https://i.imgur.com/cTbwxHZ.gif)
- Enabling Hidden gives a 30% boost to the aim bonus.
- This bonus is only given to speed if Hidden is enabled.

_Values_
Aim Values:
```
ARThresholdCeiling = 9.2
LowARThresholdCurveIntensity = 10
StarRatingThreshold = 2
BonusCeiling = 0.25
BonusCurveIntensity = 2
```

Speed Values:
```
ARThresholdCeiling = 9.2
LowARThresholdCurveIntensity = 7
StarRatingThreshold = 1.25
BonusCeiling = 0.12
BonusCurveIntensity = 3
```

_Explanation_
[Explanation of Values](https://i.imgur.com/GQmpJFC.png)

**Results**
[HONESTY +EZ 345pp -> 375pp](https://i.imgur.com/ncvt1hU.png)
[HONESTY +EZHD 404pp -> 468pp](https://i.imgur.com/dL8Pbbu.png)
[SHIORI Apex +EZDT 363pp -> 390pp](https://i.imgur.com/XI5HMDp.png)
[SHIORI Apex +EZHDDT 406pp -> 452pp](https://i.imgur.com/vPGKkyq.png)
(as you can tell, low ar stream maps gain a LOT more by adding hidden than low ar aim maps)
[Riviclia's top play: Hime Hime +EZHDDT 450pp -> 503pp](https://i.imgur.com/mNk32nq.png)
[-GN's top EZ play: Mekadesu +EZHDDT 422pp -> 473pp](https://i.imgur.com/4U4EjMD.png)
[Ekoro's top EZ play: IGNITE +EZDTFL 431pp -> 447pp](https://i.imgur.com/hglbYMG.png)
[Ekoro's second top EZ play: Daisuki Evolution +EZFL 410pp -> 465pp](https://i.imgur.com/i9hUO8O.png)
[exc's top play: d.m.c +EZHDDT 382pp -> 414pp](https://i.imgur.com/y4vY6Iq.png)

**Conclusion**
This may be considered a "bandaid fix" and that is correct, but we think that this change can possibly make the values for low AR plays a little more reasonable. We hope to get opinions from many other people, especially players with experience in this sort of stuff. Thank you for reading!

**Comment by MillhioreF** [comment]
Personally I think stream maps shouldn't be getting nearly this big of a boost on low AR since those don't become much harder to read - there's no ambiguity of the order of notes in streams. It also seems silly to buff EZFL since you're already memorizing the map at that point and there's no extra reading difficulty (in fact low AR makes FL easier in many circumstances.)

For aim maps, as well as stream maps with HD, the current proposed boost amount sounds fair to me and seems like a good step forward.

**Comment by Nyanaro** [comment]
I have to agree with MillhioreF here, using star rating a basis for reading is inherently flawed, as not only note density, but pattern density as well has no correlation with star rating. A 5 star map with spacing stretched to 7 stars isn't particularly much harder to read than that 5 star map.

**Comment by MBmasher** [comment]
Streams don't get a boost without HD as I said in the post. The pp that maps such as HONESTY gain from EZ is from aim pp.

**Comment by MillhioreF** [comment]
Makes sense. Even just the aim pp generated by streams is a bit too much right now, but fixing that is probably out of the scope of this formula and can wait for a proper reading difficulty calculation.

**Comment by diamondburned** [comment]
Based on the general idea, this means that easier difficulties played by newer players who can't play Insane or higher will have a pp bonus? I don't think rewarding it this way will help anything.

**Comment by Magnus-Cosmos** [comment]
True, but the change in pp for easier maps would be mostly unnoticeable, e.g. a [Hard] diff that gives 30pp might be buffed to 32pp.

**Comment by MBmasher** [comment]
The threshold is at 2* aim, meaning that the bonus starts at around 4-5*, and even then the AR needs to be very low for a bonus, and EVEN THEN the bonus would be extremely low.

**Comment by Karthyz** [comment]
I don't think this is the right way to go, bandaid fixes should be avoided because they just end up having to be fixed again later. Also balancing around EZ scores is probably also the wrong way to go about it because it's borderline impossible to compare EZ scores to scores without EZ because acc is basically irrelevant and if you remove the AR reduction EZ is actually far easier.

I don't know how feasible it is to look at changing and balancing EZ mod itself but I think the mod itself needs changing before looking at balancing PP around low AR because outside of EZ mod low AR is pretty much irrelevant PP wise but because EZ does make maps easier how do you possibly balance it correctly.

That aside, balancing low AR around star rating is the wrong way to go about it imo, there's so much more to it than that like others have said. Streams aren't that much harder on low AR and it's more about patterns and orders but you can't just buff jumps because some jump patterns aren't much harder on low AR also.

**Comment by Francesco149** [comment]
for me spaced streams become much more difficult to follow at low ar with HD, so I don't mind them being buffed. it's not that they're hard to read, it's just hard to follow them because there's such a big gap between where you're aiming and where the visible circles are. even just at ar8 following streams becomes my weakest point. this also applies to patterns that I can't easily snap

as for using star rating as object density... I'm not sure. some maps with really dense high bpm singletaps have pretty low star rating, like https://osu.ppy.sh/b/120030 but it might get better once star rating is more balanced
